File: Medium/Java/Math/MyPow.java
class Solution {
    public double myPow(double x, int n) {
        if (n == 0)
            return 1;
        if (n < 0) {
            n = -n; x = 1 / x;
        }
        double res = (n % 2 == 0)? myPow(x * x, n / 2): x * myPow(x * x, n / 2);
        return (res < Integer.MAX_VALUE && res > Integer.MIN_VALUE)? res: 0;
    }
}

File: Medium/Python/Math/MyPow.py
class Solution(object):
    def myPow(self, x, n):
        """
            :type x: float
            :type n: int
            :rtype: float
            """
        if n == 0:
            return 1
        return self.myPow(1/x, -n) if n < 0 else x ** n
